### PR TITLE
[new release] Five Ortac packages 0.4.0

### PR DESCRIPTION
**Pull request by n-osborne**
This release brings a number of new features and improvements:

- New features:
  + [#247](https://github.com/ocaml-gospel/ortac/pull/247): Generated tests cover now functions with multiple SUT arguments
  + [#253](https://github.com/ocaml-gospel/ortac/pull/253): Generated tests cover now functions returning SUT values
  + [#259](https://github.com/ocaml-gospel/ortac/pull/259): Generated tests can be run in a separated process with a timeout
- Improvements:
  + [#245](https://github.com/ocaml-gospel/ortac/pull/245): Fix the analysis of function signature to explicitly not support SUTs inside another type
  + [#251](https://github.com/ocaml-gospel/ortac/pull/251): Fix the display of the runnable scenario for value returned by a function that could have raised an exception

### CODE REVIEW

**Comment by mseri** [comment]
Does
[ortac-qcheck-stm.0.2.0 (failed: This expression has type ('a, STM.res) Either.t)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/ba8c37b20937faf74193844f805fb97fcc918454/variant/compilers,4.14,ortac-core.0.4.0,revdeps,ortac-qcheck-stm.0.2.0) require an upper bound on ortac 0.4.0?

File: packages/ortac-core/ortac-core.0.4.0/opam [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  "dune" {>= "3.8"}
+  "ocaml" {>= "4.12.0"}
+  "dune-build-info"
+  "dune-site"
+  "cmdliner" {>= "1.1.0"}
+  "fmt"
+  "ppxlib" {>= "0.26.0"}
+  "gospel" {= "0.3.0"}
+  "alcotest" {with-test & >= "0.8.1"}
+  "ortac-runtime" {with-test}
```
Comment (jmid) [inline packages/ortac-core/ortac-core.0.4.0/opam:35]:
```suggestion
  "ortac-runtime" {with-test & = version}
```

File: packages/ortac-dune/ortac-dune.0.4.0/opam [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+license: "MIT"
+homepage: "https://github.com/ocaml-gospel/ortac"
+bug-reports: "https://github.com/ocaml-gospel/ortac/issues"
+depends: [
+  "dune" {>= "3.8"}
+  "ocaml" {>= "4.12.0"}
+  "fmt"
+  "cmdliner" {>= "1.1.0"}
+  "ortac-core" {= version}
+  "ortac-qcheck-stm" {with-test}
```
Comment (jmid) [inline packages/ortac-dune/ortac-dune.0.4.0/opam:16]:
```suggestion
  "ortac-qcheck-stm" {with-test & = version}
```

**Review by jmid — COMMENTED** [review]
Indeed. A few of the `with-test` dependencies need to be version-constrained too, to avoid such cross-version combinations.

**Comment by n-osborne** [comment]
True. Thanks @jmid :smiley:

**Comment by jmid** [comment]
Yikes, this is still failing, e.g. with
```
ortac-core.0.4.0
  revdeps
    ortac-qcheck-stm.0.2.0 (failed: This expression has type ('a, STM.res) Either.t)
```

I think we need to separately constrain 0.2 from installing with 0.4.
In ortac-qcheck-stm/ortac-qcheck-stm.0.2.0/opam for example, I can see
```
depends: [
..
  "ortac-core" {>= "0.2.0"}
  "ortac-runtime" {with-test}
  "ortac-runtime-qcheck-stm" {with-test}
  "odoc" {with-doc}
]
```
which could benefit from `= version` constraints too!
This is probably best done in a separate PR though...

**Comment by n-osborne** [comment]
I was thinking the same and in the process of adding a commit to this PR with stricter dependencies constraints (next release of Gospel is also expected to bring breaking changes).
But I'm happy to put this changes in another PR.

**Comment by n-osborne** [comment]
#26697 should improve the situation

**Comment by jmid** [comment]
#26697 and #26702 have now been merged. Can you push a rebased version? I'm keen to see if the fixes have solved the red lights... :smiley:

**Comment by jmid** [comment]
All green and good to go!

**Comment by mseri** [comment]
Thanks!

**Pull request merged**